### ISSUE TITLE
audio ads

### ISSUE THREAD

**Issue by timstoute** [phiamo/capacitor-plugin-playlist#71]
Can anyone offer any insight into how I can incorporate support for audio ads into this plugin?  I'd like the ability to inject an audio advert into the player queue every x tracks.

Perhaps something like [Google's Interactive Media Ads SDK](https://developers.google.com/interactive-media-ads/docs/sdks/ios/client-side/background_ad_playback) would work, but the Swift side of things is out of my comfort zone.

**Comment by emilsaj** [comment]
For iOS I have opted to setting a new playlist with only the audio ad and then when the ad has finished I switch back to the original playlist with the actual audio tracks. I do this dynamically in the beginning of an audio track and in the middle of an audio track for specific tracks.

For Android this solution was not possible, since permissions are different in regards to executing commands while in background. Android will not run code after ~4 minutes of background mode, which means we cannot dynamically play ads whenever it's needed, if the app is in the background.
Only solution I found for this without too much hassle was to have the audio files for the ads be part of the queue and thus be played before the podcast and after the podcast that holds the ads, instead of in the middle of the podcast. A side effect of this approach is that users can currently skip the ads through the notification navigation since it cannot be disabled through this plugin currently. But that is not a big issue for us - especially since we have 80% of our users on iOS.

**Comment by phiamo** [comment]
yeah everything not put deeper into will e.g. lead to skippability or other issues. 
Since we are sice 0.6 on a current exo implementation, i think with a bit of effort ima could be a good solution
